src/linker/configuration.py

```diff
@@ -21,44 +21,34 @@ def __init__(
             self.computing_environment_path = None
         else:
             self.computing_environment_path = Path(computing_environment)
+        self.input_data_path = Path(input_data)
         self.pipeline = load_yaml(self.pipeline_path)
-        self.environment = self._load_computing_environment(self.computing_environment_path)
-        self.input_data = self._load_input_data_paths(Path(input_data))
         self.computing_environment = self.environment["computing_environment"]
-        self.container_engine = self.environment.get("container_engine", None)
+        self.container_engine = self.environment.get("container_engine", "undefined")
+
+    @property
+    def input_data(self) -> List[Path]:
+        file_list = [
+            Path(filepath).resolve() for filepath in load_yaml(self.input_data_path).values()
+        ]
+        missing = []
+        for file in file_list:
+            if not file.exists():
+                missing.append(str(file))
+        if missing:
+            raise RuntimeError(f"Cannot find input data: {missing}")
+        return file_list
 
     def get_resources(self) -> Dict[str, str]:
         return {
             **self.environment["implementation_resources"],
             **self.environment[self.environment["computing_environment"]],
         }
 
-    ####################
-    # Helper Functions #
-    ####################
-
-    @staticmethod
-    def _load_computing_environment(
-        computing_environment_path: Union[Path, None],
-    ) -> Dict[str, Union[Dict, str]]:
+    @property
+    def environment(self) -> Dict[str, Union[Dict, str]]:
         """Load the computing environment yaml file and return the contents as a dict."""
-        if computing_environment_path is None:
+        if self.computing_environment_path is None:
             return {"computing_environment": "local", "container_engine": "undefined"}
-        filepath = computing_environment_path.resolve()
-        if not filepath.is_file():
-            raise FileNotFoundError(
-                "Computing environment is expected to be a path to an existing"
-                f" yaml file. Input was: '{computing_environment_path}'"
-            )
+        filepath = self.computing_environment_path.resolve()
```

Comment:
This is not a great strategy if `self.environment` will be called more that once, since as written we will read from the yaml file every time this property is read.

Reply:
Will make an attribute, `self.environment = self._get_environment()`